packages/react-router-serve/bin.js

```diff
@@ -1,2 +1,7 @@
 #!/usr/bin/env node
+
+// If not already set, default `NODE_ENV=production` so React loads the proper
+// version in it's CJS entry script
+process.env.NODE_ENV = process.env.NODE_ENV ?? "production";
```

Comment:
Set this correctly before calling the CLI so the proper version of React gets loaded (https://github.com/remix-run/react-router/issues/12078)